scripts/base/protocols/dhcp/main.zeek

```diff
@@ -246,12 +251,22 @@ event DHCP::aggregate_msgs(ts: time, id: conn_id, uid: string, is_orig: bool, ms
 		{
 		# Only log the address of the server if it handed out
 		# an IP address.
-		if ( msg$yiaddr != 0.0.0.0 &&
-		     id$resp_h != 255.255.255.255 )
+		if ( msg$yiaddr != 0.0.0.0 )
 			{
-			log_info$server_addr = id$resp_h;
-			log_info$server_port = id$resp_p;
-			log_info$client_port = id$orig_p;
+			# If is_client is F and is_orig is T, it's a DHCP server
+			# broadcasting and orig_h is the server_addr, else resp_h.
```

Comment:
The code looks good, but I'm wondering if the comment is correct: we're in the server case here already (because `is_client` is false), but couldn't we get here also for a non-broadcast connection initiated by the server?

Reply:
> but couldn't we get here also for a non-broadcast connection initiated by the server?

Yep. That's the "else resp_h" part from the comment for the second if branch. Server initiated "connections" to non-broadcast will have been flipped, so server is resp_h.  Can split the commentary between the branches if that helps?

Reply:
Yep, just clarify it a bit.